install.sh

```diff
@@ -0,0 +1,321 @@
+    local actual_checksum
+    if command_exists sha256sum; then
+        actual_checksum=$(sha256sum "$archive_file" | cut -d' ' -f1)
+    elif command_exists shasum; then
+        actual_checksum=$(shasum -a 256 "$archive_file" | cut -d' ' -f1)
+    else
+        log_warning "No SHA256 tool found, skipping checksum verification"
+        return 0
+    fi
+
+    # Compare checksums
+    if [ "$actual_checksum" = "$expected_checksum" ]; then
+        log_success "Checksum verified"
+        return 0
+    else
+        log_error "Checksum verification failed!"
+        log_error "Expected: $expected_checksum"
+        log_error "Actual:   $actual_checksum"
+        return 1
+    fi
+}
+
+# Download and extract binary
+download_and_install() {
+    local os="$1"
+    local arch="$2"
+    local version="$3"
+    local install_dir="$4"
+
+    local archive_name="container-use_${version}_${os}_${arch}"
+    local extension="tar.gz"
+
+    local download_url="https://github.com/$REPO/releases/download/$version/${archive_name}.${extension}"
+    local temp_dir=$(mktemp -d)
+    local archive_file="$temp_dir/${archive_name}.${extension}"
+
+    log_info "Downloading $BINARY_NAME $version for $os/$arch..."
+
+    curl -L -o "$archive_file" "$download_url"
+
+    if [ ! -f "$archive_file" ]; then
+        log_error "Failed to download $download_url"
+        exit 1
+    fi
+
+    # Verify checksum
+    if ! verify_checksum "$archive_file" "$archive_name" "$version"; then
+        log_error "Checksum verification failed, aborting installation"
+        exit 1
+    fi
+
+    log_info "Extracting archive..."
+
+    tar -xzf "$archive_file" -C "$temp_dir"
+
+    local binary_path="$temp_dir/$BINARY_NAME"
+
+    if [ ! -f "$binary_path" ]; then
+        log_error "Binary not found in archive"
+        exit 1
+    fi
+
+    log_info "Installing to $install_dir..."
+    mkdir -p "$install_dir"
+    cp "$binary_path" "$install_dir/"
+    chmod +x "$install_dir/$BINARY_NAME"
+
+    # Clean up
+    rm -rf "$temp_dir"
+
+    log_success "$BINARY_NAME installed successfully!"
+}
+
+# Main installation process
+main() {
+    # Handle command line arguments
+    case "${1:-}" in
+        -h|--help)
+            echo "container-use installer"
+            echo ""
+            echo "Usage: $0 [options]"
+            echo ""
+            echo "Options:"
+            echo "  -h, --help    Show this help message"
+            echo ""
+            echo "This script will:"
+            echo "  1. Check for Docker installation"
+            echo "  2. Detect your OS and architecture"
+            echo "  3. Download the latest container-use binary"
+            echo "  4. Install it to your PATH"
+            exit 0
+            ;;
+    esac
+
+    log_info "Starting container-use installation..."
+
+    check_dependencies
+
+    local os=$(detect_os)
+    local arch=$(detect_arch)
+    log_info "Detected platform: $os/$arch"
+
+    local version=$(get_latest_version)
```

Comment:
It would be nice for this to be an env var, like in Dagger, to be able to specify any version

Reply:
can be added later. right now it's a pain to test that manually as there's 0 tagged releases.